plugins/aws-catalog-backend/src/BaseEntityProvider.test.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright SeatGeek
+ * Licensed under the terms of the Apache-2.0 license. See LICENSE file in project root for terms.
+ */
+import { getVoidLogger } from '@backstage/backend-common';
+import { TaskInvocationDefinition, TaskRunner } from '@backstage/backend-tasks';
+import { ResourceEntity } from '@backstage/catalog-model';
+import { EntityProviderConnection } from '@backstage/plugin-catalog-node';
+import {
+  BaseEntityProvider,
+  InstanceFilter,
+  InstanceTransformer,
+} from './BaseEntityProvider';
+
+class PersistingTaskRunner implements TaskRunner {
+  private tasks: TaskInvocationDefinition[] = [];
+
+  getTasks() {
+    return this.tasks;
+  }
+
+  run(task: TaskInvocationDefinition): Promise<void> {
+    this.tasks.push(task);
+    return Promise.resolve(undefined);
+  }
```

Comment:
nit: i think you can do this

```suggestion
  async run(task: TaskInvocationDefinition): Promise<void> {
    this.tasks.push(task);
  }
```

Reply:
Good call! Fixed via abfd75c